File: backend/src/main/java/com/github/ryand6/sudokuweb/dto/LobbyDto.java
package com.github.ryand6.sudokuweb.dto;

import com.github.ryand6.sudokuweb.enums.Difficulty;
import com.github.ryand6.sudokuweb.enums.TimeLimitPreset;
import lombok.*;

import java.time.Instant;
import java.util.Set;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
/*
LobbyDto used for transferring necessary Lobby entity fields to frontend
*/
public class LobbyDto {

    private Long id;

    private String lobbyName;

    private Difficulty difficulty;

    private TimeLimitPreset timeLimit;

    private Boolean isActive;

    private Boolean isPublic;

    private Boolean countdownActive;

    private Instant countdownEndsAt;

    private Long countdownInitiatedBy;

    private Boolean settingsLocked;

    private Set<LobbyPlayerDto> lobbyPlayers;

    private Long hostId;

}

File: frontend/src/dto/LobbyDto.ts
import type { Difficulty } from "../enum/Difficulty"
import type { TimeLimitPreset } from "../enum/TimeLimitPreset"
import type { LobbyPlayerDto } from "./LobbyPlayerDto"

export type LobbyDto = {
    id: number,
    lobbyName: string,
    difficulty: Difficulty | null,
    timeLimit: TimeLimitPreset,
    isActive: boolean,
    isPublic: boolean,
    countdownActive: boolean,
    countdownEndsAt: string,
    countdownInitiatedBy: number,
    settingsLocked: boolean,
    lobbyPlayers: LobbyPlayerDto[],
    hostId: number
}